### PR TITLE
Fixed mobile version of checkout progress bar

### LINKED ISSUE #36856: No link to shipping when in payments in checkout in mobile phone view
### Preconditions and environment

- Magento version 2.4.5
- Anything else that would help a developer reproduce the bug
Luma theme

### Steps to reproduce

Use a mobile phone or mobile view in browser dev tools
1. Add an item to checkout
1. Go to checkout
1. Fill in the shipping data and click next
1. Being in payment mode there is no way to go back. There is no link to shipping. The checkout titles/links "Shipping" and "Review & Payments" are hidden from the top of the page.

### Expected result

The checkout titles/links "Shipping" and "Review & Payments" should always be visible on top of the page to help users know at which step they are and to allow corrections on the shipping when being in payments.

### Actual result

The checkout titles/links "Shipping" and "Review & Payments" are hidden from the top of the page.
If you rotate your phone or widen the screen in dev tools in browser they show up.

### Additional information

*No response*

### Release note

*No response*

### Triage and priority

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [ ] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [X] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, "quality" or "usability".*

### PR DESCRIPTION
**Pull request by nidhigupta13-ey**
### Description (*)

Fixed the issue in mobile for checkout progress bar. Now it's showing in mobile and tablet view.

### Related Pull Requests

https://github.com/magento/magento2/issues/36856

### Fixed Issues (if relevant)

1. Fixes magento/magento2#36856

### Manual testing scenarios (*)

1. We need to test it in mobile view.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] README.md files for modified modules are updated and included in the pull request if any [README.md predefined sections](https://github.com/magento/devdocs/wiki/Magento-module-README.md) require an update
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by nidhigupta13-ey** [comment]
@magento run all tests

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less [hunk 1]
```diff
@@ -57,8 +60,6 @@
 //  Desktop
 //  _____________________________________________
```
Comment (mrtuvn) [inline app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:57]: any reason of remove this media mixin ?

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less [hunk 2]
```diff
@@ -195,4 +196,27 @@
             }
         }
     }
+
+//
+// Mobile max width 600px
+//
```
Comment (mrtuvn) [inline app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:198]: should target for mobile device. These lines will also maybe duplicated for desktop

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less (hunk 1 above)
Comment (nidhigupta13-ey) [reply app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:57]: As this issue is working fine in blank theme. So I compared the code and there as well we don't have any mixin and using the progress bar code irrespective of view. So the progress bar will show in each device.

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less (hunk 2 above)
Comment (nidhigupta13-ey) [reply app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:198]: It is not duplicated. Some kind of thing like margin and width is only changing based on screen size.

Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:198]: without wrap between media mixins your code styles probably duplicated in both final output files styles-m.css and styles-l.css as i remember before
That's why code base use mixins for target mobile or desktop

Comment (mrtuvn) [reply app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:198]: If you want apply for mobile move code between for apply also mobile and desktop
```less 
& when (@media-common = true) {
//Your code here
}
```

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less [hunk 3]
```diff
@@ -195,4 +196,27 @@
...
+    }
+}
+
+//
+// Mobile max width 320px and 280px
+//
+@media (min-width: 280px) and (max-width: 320px) {
+    .opc-progress-bar-item {
+        .lib-css(width, @checkout-progress-bar-item__width__mobile_small);
+    }
```
Comment (mrtuvn) [inline app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:216]: Also you should use screen variables instead hardcoded fix value

**Comment by cptX** [comment]
Is there any final solution to this problem yet? 

Should I now copy the _progress-bar.less file from the dev to my 2.4.6 installation? Will it work or do you need further tests?

**Comment by nidhigupta13-ey** [comment]
@cptX  app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less this is the path of the file. You can copy your file in this path.

**Comment by cptX** [comment]
Hi @nidhigupta13-ey ,
I placed the file to my custom theme but I didn't see any change

app/design/frontend/MyName/MyCustomTheme/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less

Should I place it exactly with the "Magento/luma" path or can it work with the above path as well? If it should work with my custom theme then unfortunately it didn't work...

**Comment by nidhigupta13-ey** [comment]
@cptX try to place it with "Magento/luma" path

**Comment by cptX** [comment]
> @cptX try to place it with "Magento/luma" path

It didn't work even in Magento/luma path... I cannot see the progress bar in the checkout page in mobile...

**Comment by cptX** [comment]
Also I see that in checkout in mobile the menu button (3 horizontal lines) is also missing. Is this expected? Does it make sense?

**Comment by nidhigupta13-ey** [comment]
@cptX those three lines are not there in desktop view as well, so I don't think so it should be there in mobile view.

**Comment by cptX** [comment]
OK regarding the Menu lines just ignore it. I still believe that the progress bar is not visible in mobile after the patch. Whatever I tried didn't work.
Is there any other workaround?

**Comment by nidhigupta13-ey** [comment]
@cptX  I have tried the same in my another magento instance and it is working for me. Can you please try to clear cache again and try in incognito if it is working or not? You can also check in pub csc the changes are reflecting or not.

**Comment by cptX** [comment]
Can you please tell me what path should I use for the patch if I create a custom luma theme?
Putting the patch in the "../frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less" should work also for a custom luma theme in a separate folder? Or is it better to patch directly the path in the core?

**Comment by cptX** [comment]
I ran extensive tests in my original luma theme and in my custom luma theme and is impossible to make it work.
I placed the file in this location
app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less
Steps are not visible at all in mobile...
Can you please check once more the path you suggested me?

**Comment by nidhigupta13-ey** [comment]
@cptX I don't know why your changes are not working. But for me it is working fine in the given path, Can you please check your changes are there in style-m or style-l files.

**Comment by cptX** [comment]
I finally found the correct location to put the file in order to work

app/design/frontend/Vendor/Customtheme/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less

**Comment by nidhigupta13-ey** [comment]
@cptX happy to know that you found the path and it is working..

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less [hunk 4]
```diff
@@ -137,12 +133,12 @@
                 content: counter(i);
                 counter-increment: i;
                 .lib-typography(
```
Comment (mrtuvn) [inline app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:135]: you should clean up this properties of lib mixin to inline instead break every param into new line

File: app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less [hunk 5]
```diff
@@ -196,3 +192,27 @@
...
+// Mobile
+//
+.media-width(@extremum, @break) when (@extremum = 'max') and (@break = @screen__s) {
+    .opc-progress-bar {
+        .lib-css(margin, 0 0 @checkout-progress-bar__margin__mobile);
+        counter-reset: i;
+        display: block;
+        font-size: 0;
+    }
+    .opc-progress-bar-item {
```
Comment (mrtuvn) [inline app/design/frontend/Magento/luma/Magento_Checkout/web/css/source/module/checkout/_progress-bar.less:205]: need whitespace between each selectors

**Comment by nidhigupta13-ey** [comment]
@mrtuvn I have updated the code and pushed again.

**Comment by cptX** [comment]
Can you tell me if this is resolved in the stable release 2.4.7?

**Comment by cptX** [comment]
I don't know why the progress has stopped on this? Isn't this high priority? I mean having a progress bar in checkout in mobile isn't of high importance and functionality??

**Comment by engcom-Hotel** [comment]
@magento run all tests

**Review by engcom-Hotel — APPROVED** [review]

**Comment by engcom-Bravo** [comment]
Hi @nidhigupta13-ey,

Thanks for the collaboration & contribution!

### :heavy_check_mark: QA Passed


**Preconditions:**

- Install fresh Magento **`2.4-develop`**

**Steps to reproduce**

Use a mobile phone or mobile view in browser dev tools

- Add an item to checkout
- Go to checkout
- Fill in the shipping data and click next
- Being in payment mode there is no way to go back. There is no link to shipping. The checkout titles/links "Shipping" and "Review & Payments" are hidden from the top of the page.

**Before:** :heavy_multiplication_x: 


Screenshot 2024-11-15 at 10 10 55



**After:** :heavy_check_mark: 

Screenshot 2024-11-15 at 10 16 36



Screenshot 2024-11-15 at 10 17 26






Builds are failed. Hence, moving this PR to **Extended Testing**.

Thanks.

**Comment by engcom-Dash** [comment]
@magento run Unit Tests, Functional Tests CE, Functional Tests B2B

**Comment by engcom-Dash** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by engcom-Dash** [comment]
@magento run Functional Tests CE, Functional Tests B2B

**Comment by engcom-Dash** [comment]
@magento run all tests

**Comment by engcom-Dash** [comment]
@magento run Integration Tests, Functional Tests CE, Functional Tests B2B

**Comment by engcom-Dash** [comment]
The Functional B2B test failures are different in recent 2 successive builds. The Functional CE test failures seem flaky, and also there is a known [issue](https://jira.corp.adobe.com/browse/ACP2E-3481). They neither part of PR nor failing because of the PR changes, hence moving it to Merge in Progress.

**Functional Tests B2B**
Run 1: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/36982/b63f36644d179449d5690be24f7735ad/Functional/allure-report-b2b/index.html#categories/16b25a50b4a8b68f5548cdba82ebd5f8/de373c800213d2be/
image

Run 2: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/36982/d512d9162aae25676ec5dd94a783de50/Functional/allure-report-b2b/index.html#categories/8fb3a91ba5aaf9de24cc8a92edc82b5d
image

**Functional Tests CE**
Run 1: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/36982/2fa9409bcd83ab6ed09f7dfdb9a394a3/Functional/allure-report-ce/index.html#categories/6d7e37752c3cc83cb3d2063131f608bb/c8b8c08819c056c6/
image

Run 2: https://public-results-storage-prod.magento-testing-service.engineering/reports/magento/magento2/pull/36982/13b5b25017d3d6ea6c68076fbe708cbe/Functional/allure-report-ce/index.html#categories/8fb3a91ba5aaf9de24cc8a92edc82b5d
image

**Pull request merged**